ReportApp.Core/Repository/ReportRepository.cs

```diff
@@ -97,5 +98,15 @@ private static byte[] ConvertToBytes(HttpPostedFileBase file)
             return fileBytes;
         }
 
+        public void RemoveAttachedFile(int? reportId)
+        {
+            AttachedFile file = _context.AttachedFiles.FirstOrDefault(x => x.ReportId == reportId);
+            if (file != null)
+            {
+                var fileToDelete = _context.AttachedFiles.Where(x => x.ReportId == reportId);
```

Comment:
cache _context.AttachedFiles in a variable since you are reusing it in 3 places.